app/Models/Image.php

```diff
@@ -124,27 +115,7 @@ public static function upload(UploadedFile $file, $user, bool $forceReupload = f
 
             $image->ensureDirectoryExists();
             foreach (self::$ImageTypes as $coverType) {
-                if ($coverType['id'] === self::ORIGINAL && $image->mime === 'image/jpeg') {
-                    $command = 'cp "'.$file->getPathname().'" '.$image->getFile($coverType['id']);
-                } else {
-                    // ImageMagick options reference: http://www.imagemagick.org/script/command-line-options.php
-                    $command = 'convert 2>&1 "'.$file->getPathname().'" -background white -alpha remove -alpha off -strip';
-
-                    if ($image->mime === 'image/jpeg') {
-                        $command .= ' -quality 100 -format jpeg';
-                    } else {
-                        $command .= ' -quality 95 -format png';
-                    }
-
-                    if (isset($coverType['width']) && isset($coverType['height'])) {
-                        $command .= " -thumbnail ${coverType['width']}x${coverType['height']}^ -gravity center -extent ${coverType['width']}x${coverType['height']}";
-                    }
-
-                    $command .= ' "'.$image->getFile($coverType['id']).'"';
-                }
-
-                External::execute($command);
-                chmod($image->getFile($coverType['id']), 0644);
+                self::processFile($file, $image->getFile($coverType['id']), $coverType);
```

Comment:
This looks like it should be wrapped in a try-catch and have some exception handling going on. I realize you're adapting old code that may not have originally had any but it's a good opportunity to add some.

Reply:
I just realized this already is wrapped in a try-catch. Disregard this comment. Sorry!